kotlin-assert-utils/src/main/kotlin/se/lovef/assert/assertUtil.kt

```diff
@@ -164,3 +166,16 @@ infix fun String.proof(proof: () -> Unit) {
         throw AssertionError(this, throwable)
     }
 }
+
+fun eventually(block: () -> Unit) {
+    var thrown: Throwable? = null
+    thread {
+        try {
+            Thread.sleep(100)
```

Comment:
* Parameterize maximum duration 
* Make several test with some interval up to maximum duration